### PR TITLE
Adds TF configuration to create a Kafka topic and producer for invoice-fulfillment service

### PR DESCRIPTION
**Pull request by agkountis**
This PR adds TF configuration to create a new Kafka topic and a producer on the new Kafka cluster using the module approach.

### CODE REVIEW

File: dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+resource "kafka_topic" "invoice-fulfillment" {
+  name               = "bex.internal.bill_fulfilled"
```
Comment (sbuliarca) [inline dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf:2]: Will we need migration of the data ? Especially for prod?

If we do, we must keep the name of the current topic + a namespace prefix, as we'll use mirror maker 2 to migrate the data.
I guess the namespace prefix is `bex` atm.

File: dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    # keep data for 7 days
+    "retention.ms" = "604800000"
+    # allow max 1 MB for a message
+    "max.message.bytes" = "1048576"
+    "compression.type" = "zstd"
+    "cleanup.policy"   = "delete"
+  }
+}
+
+module "producer" {
```
Comment (sbuliarca) [inline dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf:15]: Do you need also a consumer? Or you'll add it later in another PR?

File: dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf
Comment (sbuliarca) [inline dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf:1]: The way TF works, we need to keep all the files in the kafka-shared folder. So you'll need to use some prefixing in the name of the file to give it more meaning

File: dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf (hunk 2 above)
Comment (sbuliarca) [inline dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf:15]: I think this should have a more meaningful name, like module "invoice_fulfillment_producer"

File: dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf (hunk 1 above)
Comment (espang-uw) [reply dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf:2]: This is luckily a new topic for a new system, so we don't have to migrate anything.

Our services run in the `customer-billing` namespace, would it be okay if we consistently take the `bex` prefix or should this be a match with the k8s-namespace?

File: dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf
Comment (agkountis) [reply dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf:1]: Got it. Changing this now.

File: dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf (hunk 2 above)
Comment (agkountis) [reply dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf:15]: Assumed this was referencing something else and required the same name as the example. I will change it to the more descriptive version.

Comment (agkountis) [reply dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf:15]: For now the consuming service implementation has not started, so will add a consumer at a later date on a different PR.

File: dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf (hunk 1 above)
Comment (sbuliarca) [reply dev-aws/kafka-shared/customer-billing/invoice-fulfillment/invoice-fulfillment.tf]: I think bex should be fine

**Review by sbuliarca — APPROVED** [review]

**Pull request merged**